Put examples by both ruby lib and ruby lib core

KazuCocoa opened the pull request:
## Proposed changes

Would like to set examples both ruby_lib and ruby_lib_core since I got questions about it, sometimes


## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/appium/appium/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla.js.foundation/appium/appium)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

KazuCocoa committed:
add # ruby_lib_core

KazuCocoa committed:
add ruby_lib examples

imurchie commented:
Would we be better served having two different sections (like we do for Javascript)?

imurchie commented:
Also, does documentation for each of these live in different places? Do the links need to be updated?

KazuCocoa commented:
ruby_lib wraps ruby_core_lib. Almost commands for webdriver/Appium are only in the core. So, almost links should be same. 
I’ll make sure links as well

KazuCocoa committed:
rename ruby to ruby_lib_core [ci skip]

imurchie commented on commands-yml/commands/mobile-command.yml:
```diff
@@ -92,6 +92,11 @@
+        # ruby_lib
+        execute_script("mobile: scroll", { "direction" => "down"})
+
+  ruby_lib_core:
```
Indentation?

imurchie commented on commands-yml/commands/context/get-context.yml:
```diff
@@ -27,8 +27,10 @@ example_usage:
     |
       let context = driver.context();
-  ruby:
+  ruby_lib:
```
What is the intended difference between `ruby_lib` and `ruby` (which is what was there, and remains starting with `commands-yml/commands/device/app/get-app-strings.yml` in this PR)?

imurchie commented on commands-yml/commands/device/app/close-app.yml:
```diff
@@ -14,7 +14,11 @@ example_usage:
-  ruby:
+  ruby_lib:
+    |
+      # ruby_lib
```
These comments are not necessary.

imurchie commented:
There is a further need to adjust the template, as both `ruby_lib` and `ruby_lib_core` will be ignored as is.

This diff should suit if `ruby`, `ruby_lib`, and `ruby_lib_core` are all valid (note that the internal three-backticks had to be manipulated (a space added in the middle) so that they would not get rendered as starts and stops of the diff block):
```diff
diff --git a/commands-yml/template.md b/commands-yml/template.md
index 8046b6ab..67c8125c 100644
--- a/commands-yml/template.md
+++ b/commands-yml/template.md
@@ -42,12 +42,27 @@
 ` ``

 ` ``ruby
-{{#if example_usage.ruby}}
 # Ruby
+# ruby example
+{{#if example_usage.ruby}}
 {{example_usage.ruby}}
 {{else}}
 # Not supported
 {{/if}}
+
+# ruby_lib example
+{{#if example_usage.ruby_lib}}
+{{example_usage.ruby_lib}}
+{{else}}
+# Not supported
+{{/if}}
+
+# ruby_lib_core example
+{{#if example_usage.ruby_lib_core}}
+{{example_usage.ruby_lib_core}}
+{{else}}
+# Not supported
+{{/if}}
 ` ``

 ` ``php
```

And the following should work if `ruby` is omitted:
```diff
diff --git a/commands-yml/template.md b/commands-yml/template.md
index 8046b6ab..36b77808 100644
--- a/commands-yml/template.md
+++ b/commands-yml/template.md
@@ -42,9 +42,17 @@
 ` ``

 ` ``ruby
-{{#if example_usage.ruby}}
 # Ruby
-{{example_usage.ruby}}
+# ruby_lib example
+{{#if example_usage.ruby_lib}}
+{{example_usage.ruby_lib}}
+{{else}}
+# Not supported
+{{/if}}
+
+# ruby_lib_core example
+{{#if example_usage.ruby_lib_core}}
+{{example_usage.ruby_lib_core}}
 {{else}}
 # Not supported
 {{/if}}
```

imurchie commented:
This PR involves things that _are_ in the build, so CI should be run. You can check that things render ok by running `npm run generate-docs` locally.

KazuCocoa committed:
add doc for ruby_core, tweak library names

KazuCocoa committed:
fix fialing to generate doc

KazuCocoa committed:
update template.md for ruby_lib_core

KazuCocoa committed:
add a url of ruby_lib_core

KazuCocoa commented:
Updated. The doc will be like https://github.com/appium/appium/pull/11694/files#diff-4f25536461d7b7628f212fea4690a028L44
I've also updated https://github.com/appium/ruby_lib_core and https://github.com/appium/ruby_lib to address each other.
